clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1228,6 +1270,17 @@ private int partition(ProducerRecord<K, V> record, byte[] serializedKey, byte[]
                         record.topic(), record.key(), serializedKey, record.value(), serializedValue, cluster);
     }
 
+    private void throwIfInvalidGroupMetadata(ConsumerGroupMetadata groupMetadata) {
+        if (groupMetadata == null) {
+            throw new IllegalStateException("Consumer group metadata could not be null");
```

Comment:
I think these should all be `IllegalArgumentException`. The producer is not in an illegal state.